src/react/DeviceInterfaceValues.js

```diff
@@ -16,229 +16,122 @@
    limitations under the License.
 */
 
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { Card, Container, Spinner, Table } from "react-bootstrap";
 
 import Device from "./astarte/Device.js";
 import BackButton from "./ui/BackButton.js";
+import WaitForData from "./components/WaitForData.js";
+import useFetch from "./hooks/useFetch.js";
+import { useAlerts } from "./AlertManager";
+
+const MAX_VALUES = 20;
+
+const DeviceInterfaceValues = ({ astarte, hisory, deviceId, interfaceName }) => {
+  const [ interfaceType, setInterfaceType ] = useState(null);
+  const deviceData = useFetch(() => astarte.getDeviceData({
+    deviceId: deviceId,
+    interfaceName: interfaceName,
+  }));
+
+  const deviceAlerts = useAlerts();
+
+  useEffect(() => {
+    const getInterfaceType = async () => {
+      const deviceInfos = await astarte.getDeviceInfo(deviceId)
+        .catch(() => { throw new Error('Device not found.'); });
```

Comment:
You are throwing a bunch of Errors inside `getInterfaceType` which will not be caught. Maybe they should trigger some alerts